src/Features/Core/Portable/CodeRefactorings/IRefactoringHelpersService.cs

```diff
@@ -17,7 +17,9 @@ internal interface IRefactoringHelpersService : ILanguageService
         /// <para>
         /// A <typeparamref name="TSyntaxNode"/> instance is returned if:
         /// - Selection is zero-width and inside/touching a Token with direct parent of type <typeparamref name="TSyntaxNode"/>.
-        /// - Selection is zero-width and touching a Token whose ancestor ends/starts precisely on current selection .
+        /// - Selection is zero-width and touching a Token whose ancestor of type <typeparamref name="TSyntaxNode"/> ends/starts precisely on current selection.
+        /// - Selection is zero-width and in whitespace that corresponds to a Token whose direct ancestor is of type of type <typeparamref name="TSyntaxNode"/>.
+        /// - Selection is zero-width and in a header (defined by ISyntaxFacts helpers) of an node of type of type <typeparamref name="TSyntaxNode"/>.
```

Comment:
so plan would be sharing it in all referactoring get node? even thought it is only 1 place used right now?

Reply:
I'm not sure I follow the question.

Reply:
If the question was about the overload that took Context, that got replaced by a set of Extension methods by Cyrus. Feel free to reopen if I'm wrong.

Reply:
The TrySelectedNode is used only by one place because cyrus prepared a helpful extension method helper on Context (and that one is used by a number (> 10) of refactorings already)